clippy_lints/src/utils/sugg.rs

```diff
@@ -162,10 +162,10 @@ impl<'a> Sugg<'a> {
 
     /// Convenience method to create the `<lhs>..<rhs>` or `<lhs>...<rhs>`
     /// suggestion.
-    pub fn range(self, end: Self, limit: ast::RangeLimits) -> Sugg<'static> {
+    pub fn range<E: Borrow<Self>>(self, end: E, limit: ast::RangeLimits) -> Sugg<'static> {
```

Comment:
Any reason not to simply use `&Self`?

Reply:
I thought it could be better for these combinator-like things, and was too lazy to look for callsites...

Reply:
> too lazy to look for callsites...

I can totally get behind that :D